### DISCUSSION TITLE
Modify Export MD File Name Template to replace hyphens with spaces

### DISCUSSION THREAD

**Discussion by jojo-sc** [windingwind/zotero-better-notes#1164] [Q&A] [answered]
Hi there,

I am trying to export and sync md files to obsidian. My literature notes only have names like "Müller 2019". I have modified the export template to not include the item key, this leaves me with hyphens in the name, e.g. "-Müller-2019-". How can I change the template so that the hyphens are replaced by spaces, so that the name can be "Müller 2019"? Is this possible? I am not familiar with the programming language used. If I manually change the name in obsidian, the sync stops, even though the item key is mentioned in the yaml data.

Thanks!

**Comment by windingwind** [comment, marked as answer]
Any non-alphabet character will be converted to `-` in the filename.

Btw, I don't really suggest you use this format as the file name. If you have two notes under the same regular item, there will be two `author-year.md` and they will try to overwrite each other.

**Reply by jojo-sc** [reply]
Thanks for your quick reply. In my workflow, I only want to export one note per item anyhow, so I don't believe there will be issues there. I will work around it by using aliases to link to my literature notes. Thank you for your work, I really like the plugin :-)

**Reply by windingwind** [reply]
I just tried myself and actually, you can export to the format you want. The characters won't be converted if you don't write the conversion in the filename template.

Try this:


```
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Tools->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
name: "[ExportMDFileNameV2]"
zoteroVersion: "7.0.9-beta.3+ece595e79"
pluginVersion: "2.0.18"
savedAt: "2024-10-25T16:28:13.167Z"
content: |-
  ${{
    const topItem = noteItem.parentItem;
    if (topItem) {
  	return `${topItem.firstCreator} ${topItem.getField('year')}.md`;
    }
    return `${noteItem.getNoteTitle()}-${noteItem.key}.md`;
  }}$
```
